src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1596,6 +1617,8 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
             INuGetProjectContext nuGetProjectContext,
             IReadOnlyCollection<SourceRepository> activeSources,
             VersionRange versionRange,
+            string newMappingID,
+            string newMappingSource,
```

Comment:
> The newMappingID and newMappingSource pair are related. I propose the introduction of a new type that enforces the relationship between these two and encapsulates the logic of what happens when one is null and not the other?

I agree it could be more convenient to have a type, especially in the future if we want to support more configurable properties from the client. However, I think it's safe to iterate on this since the API isn't one that we ship, and as far as I know nothing else in VS depends on this API. In other words, we don't have to minimize our changes to this API right now, and when we have more interesting capabilities we want to support that warrant it, then we can create and swap this over to a new type.

> Currently, I am not seeing a test that checks what will happen if only one of these strings is null and the other is not.

The source code only uses them if both are not null. I'm not sure a separate test for one being null and not the other would be a useful test.

There's a test for this code path with both being `null` (which exercises the same code as only 1 being `null`): https://github.com/NuGet/NuGet.Client/blob/5e03c332991fcf02574f9e8b7a3eb7918aea91a5/test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs#L7111-L7111

and there's a test for both being not null, ensuring it affects the restore summary:
https://github.com/NuGet/NuGet.Client/blob/5e03c332991fcf02574f9e8b7a3eb7918aea91a5/test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs#L7182-L7182